### PR TITLE
TestCommitEpoch unit test

### PR DESCRIPTION
**Pull request by Nemanja0x**
# Description
An unit test which add validators, connects them with delegators and commits an epoch.
It is possible to get gas used  for committing an epoch

# Changes include

- [ ] Bugfix (non-breaking change that solves an issue)
- [ ] Hotfix (change that solves an urgent issue, and requires immediate attention)
- [x] New feature (non-breaking change that adds functionality)
- [ ] Breaking change (change that is not backwards-compatible and/or changes current functionality)

# Checklist

- [x] I have assigned this PR to myself
- [x] I have added at least 1 reviewer
- [ ] I have added the relevant labels
- [ ] I have updated the official documentation
- [ ] I have added sufficient documentation in code

## Testing

- [ ] I have tested this code with the official test suite
- [x] I have tested this code manually

### CODE REVIEW

**Review by igorcrevar — APPROVED** [review]

File: consensus/polybft/checkpoint_manager_test.go [hunk 1]
```diff
@@ -582,6 +585,157 @@ func TestPerformExit(t *testing.T) {
...
+
+		// get data for ChildValidatorSet initialization
+		initInput, err := getInitChildValidatorSetInput(polyBFTConfig)
+		require.NoError(t, err)
+
+		// init ChildValidatorSet
+		err = initContract(contracts.ValidatorSetContract, initInput, "ChildValidatorSet", transition)
+		require.NoError(t, err)
+
+		// connect validators and delegators
```
Comment (goran-ethernal) [inline consensus/polybft/checkpoint_manager_test.go]: Nitpick - maybe change this comment to "delegate amounts to validators"

File: consensus/polybft/helpers_test.go [hunk 2]
```diff
@@ -19,6 +19,18 @@ func createTestKey(t *testing.T) *wallet.Key {
 	return wallet.NewKey(wallet.GenerateAccount())
 }
 
+func createRandomTestKeys(t *testing.T, amount int) []*wallet.Key {
```
Comment (goran-ethernal) [inline consensus/polybft/helpers_test.go]: Nitpick - maybe call this `amount` parameter - `numberOfKeys`.

**Review by goran-ethernal — APPROVED** [review]
Except for nitpicks, LGTM :100:

File: consensus/polybft/checkpoint_manager_test.go (hunk 1 above)
Comment (Nemanja0x) [reply consensus/polybft/checkpoint_manager_test.go]: Fixed https://github.com/0xPolygon/polygon-edge/pull/1167/commits/a2201f5a6ab19aa537d88ba92ceac9806bdbca5a

File: consensus/polybft/helpers_test.go (hunk 2 above)
Comment (Nemanja0x) [reply consensus/polybft/helpers_test.go]: Fixed https://github.com/0xPolygon/polygon-edge/pull/1167/commits/a2201f5a6ab19aa537d88ba92ceac9806bdbca5a

**Pull request merged**